src/TypeSpec.Extension/Emitter.Csharp/src/options.ts

```diff
@@ -129,7 +129,7 @@ const defaultOptions = {
     generateTestProject: true
 };
 
-export function resolveOptions(context: EmitContext<NetEmitterOptions>) {
+export function resolveOptions(context: EmitContext<any>) {
```

Comment:
Personally I don't like it. I'll let others to give opinion.

Reply:
This is because `sdkContext.emitContext` is `EmitContext<Record<string, any>>`, which does not comply with `EmitContext<NetEmitterOptions>`

Reply:
And we can't change it to `EmitContext<NetEmitterOptions | Record<string, any>>`, because some properties only exist in NetEmitterOptions, it will cause compile error.
Another option is to have a different method accepting different types, which would cause duplicate code...

Reply:
Huh? Why [sdkContext.emitContext](https://github.com/Azure/typespec-azure/blob/0a5020b2964d7fc565ddc42ad09147b4b22cf3f4/packages/typespec-client-generator-core/src/interfaces.ts#L34) is `EmitContext<Record<string, any>>` ?  It should be `EmitContext<NetEmitterOptions>`

Reply:
Because [`NetEmitterOptions`](https://github.com/Azure/autorest.csharp/blob/4e2f9487d2b08f1909a1d520c70b38052ca2056e/src/TypeSpec.Extension/Emitter.Csharp/src/options.ts#L7) is defined by us, not in tcgc.

Reply:
set generic type for SdkContext as `SdkContext<NetEmitterOptions>` could work

Reply:
forgot to revert this, fixed now